File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  // takes in a string and returns it in reverse order
  // creates a an empty array called result
  // split the input string into an array of elements
  // iterate through the array from the end to the beginning using a reverse for loop pushing each element into the results array
  // join the reversed array
  // return the joined reversed array which is now a string

  let strArr = str.split('');
  let result =[]
  for (i = strArr.length - 1; i >=0; i--) {
    result.push(strArr[i]);
    // console.log("reverse elements:", strArr[i]);
  }
  return result.join('');
}


if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("Expecting: ''", reverseString(""));

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));


}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  # use .chars method to turn string into an array
  # use the .reduce mthod which will reduce each element of an enumerable into a single value
  # reverse and concat x and y variables

  str.chars.reduce { |x,y| y + x }

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts "Expecting: 'nhoJ olleH'"
  puts "=>", reverse_string('Hello John')

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution